Sources/App/Extensions/Vapor.Request+RequestProtocol.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+import Bot
+import Vapor
+
+extension Vapor.Request: RequestProtocol {
+    public func header(named name: String) -> String? {
+        return http.headers["X-GitHub-Event"].first
+    }
+
+    public func decodeBody<T>(_ type: T.Type) -> T? where T: Decodable {
+        return try? content.syncDecode(type)
```

Comment:
this shouldn't be used - in Vapor 3.3.0 it results in a runtime error (not crash). should use `content.decode` and chain futures 🤷‍♂ everything is the Future when dealing with Vapor

Reply:
I'd like to avoid dealing with them for now if possible.

Reply:
The issue is that it's not safe in SwiftNIO environment, it can potentially deadlock (I don't know what is the real scenario for this to happen though). We have everything based on futures in Stevenson and it's fine, even though technically most of the performed tasks don't have to be asynchronous.